From 04fdb0a90b9371eb354b6079fcd25bcea1aa5e11 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint.reczey@canonical.com>
Date: Sat, 9 Jan 2021 08:35:28 +0100
Subject: [PATCH] tests: Omit some checks for msgctl and shmctl where addresses
 are bogus

The false failures occur with armhf binaries running on arm64 kernel.

* tests/ipc_msg.c
* tests/ipc_shm.c
---
 tests/ipc_msg.c | 2 ++
 tests/ipc_shm.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/tests/ipc_msg.c b/tests/ipc_msg.c
index f5a596227..857b09276 100644
--- a/tests/ipc_msg.c
+++ b/tests/ipc_msg.c
@@ -151,8 +151,10 @@ print_msqid_ds(const char *const str_ipc_cmd,
 	       const int rc)
 {
 	if (rc < 0) {
+#if TEST_MSGCTL_BOGUS_ADDR
 		printf("msgctl\\(%d, (%s\\|)?%s, %p\\) = %s\n",
 		       id, str_ipc_64, str_ipc_cmd, ds, sprintrc_grep(rc));
+#endif
 		return;
 	}
 
diff --git a/tests/ipc_shm.c b/tests/ipc_shm.c
index 99c292911..facf1fed6 100644
--- a/tests/ipc_shm.c
+++ b/tests/ipc_shm.c
@@ -121,8 +121,10 @@ print_shmid_ds(const char *const str_ipc_cmd,
 	       const int rc)
 {
 	if (rc < 0) {
+#if TEST_SHMCTL_BOGUS_ADDR
 		printf("shmctl\\(%d, (%s\\|)?%s, %p\\) = %s\n",
 		       id, str_ipc_64, str_ipc_cmd, ds, sprintrc_grep(rc));
+#endif
 		return;
 	}
 	printf("shmctl\\(%d, (%s\\|)?%s, \\{shm_perm=\\{uid=%u, gid=%u"
-- 
2.25.1

